### PR TITLE
Add OpenBSD Support

### PR DESCRIPTION
**Pull request by LainLayer**
Minor nob modifications to support compiling on OpenBSD.
Hot reloading seems to work, so does rendering. 

The audio when playing an audio file sounds very slowed down, too bad, I don't know how to fix it. Everything else works :)

### CODE REVIEW

**Comment by LainLayer** [comment]
It seems the messed up audio is a general Raylib/miniaudio phenomena on OpenBSD and not a fault in the musializer code. Considering no one cares about OpenBSD, it will likely take a while until it is fixed. Until then it at least compiles and runs.

File: src/nob_openbsd.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                nob_cmd_append(&cmd,
+                    "-Wl,-rpath=./build/",
+                    "-Wl,-rpath=./",
+                    nob_temp_sprintf("-Wl,-rpath=./build/raylib/%s", MUSIALIZER_TARGET_NAME),
+                    // NOTE: just in case somebody wants to run musializer from within the ./build/ folder
+                    nob_temp_sprintf("-Wl,-rpath=./raylib/%s", MUSIALIZER_TARGET_NAME));
+                nob_cmd_append(&cmd,
+                    nob_temp_sprintf("-L./build/raylib/%s", MUSIALIZER_TARGET_NAME),
+                    "-l:libraylib.so");
+                nob_cmd_append(&cmd, "-lm", "-lpthread");
```
Comment (rexim) [inline src/nob_openbsd.c:47]: The only significant difference between Linux and OpenBSD seems to be the lack of `-ldl` in OpenBSD. Why so? Does linking with `-ldl` create some sort of problems? Or is it not possible at all?

**Comment by ismagilli** [comment]
On FreeBSD `dl*` functions live in `libc`. On linux `ld*` functions placed in a separate library (`libdl`). I think the situation with OpenBSD is similar to the situation with FreeBSD. Possible solution: `ln -s /usr/lib/libc.so /usr/lib/libdl.so` as root. @LainLayer It seems to me that it is better to use the same implementation of nob on linux and BSD, put the `-ldl` flag in a separate call to the `nob_cmd_append` function and wrap it in ifdef-endif. And maybe rename target from `LINUX` to `UNIX`.


BSD detection (see [boost/bsd.hpp](https://www.boost.org/doc/libs/1_83_0/boost/config/platform/bsd.hpp)):
```cpp
#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
// Your code here...
#endif
```

File: src/nob_openbsd.c (hunk 1 above)
Comment (ismagilli) [reply src/nob_openbsd.c:47]: @rexim `dl*` functions on FreeBSD is a part of `libc`. And there is no library with name `libdl`. I think on OpenBSD situation is same.

**Comment by ismagilli** [comment]
Note that this is only MY opinion. It is better to wait **rexim** reply.

File: src/nob_openbsd.c (hunk 1 above)
Comment (rexim) [reply src/nob_openbsd.c:47]: I see! Thank you!

**Comment by rexim** [comment]
@ismagilli I was thinking about making a more general POSIX builder sort of speak. But I'll do it myself later. 

Thank you for the contribution guys!

File: src/nob_openbsd.c (hunk 1 above)
Comment (LainLayer) [reply src/nob_openbsd.c:47]: you also need to add `-I/usr/X11R6/include` since OpenBSD puts its X11 sources in another place.

**Pull request merged**